packages/components/drawer/__tests__/drawer.test.tsx

```diff
@@ -0,0 +1,356 @@
+  })
+
+  test('render correct content', async () => {
+    const wrapper = mount(() => (
+      <Drawer title={title} modelValue={true}>
+        <span>{content}</span>
+        <Button>cancel</Button>
+        <Button type="primary">confirm</Button>
+      </Drawer>
+    ))
+
+    await nextTick()
+    expect(wrapper.find('.el-drawer__body span').element.textContent).toEqual(
+      content
+    )
+    const footerBtns = wrapper.findAll('.el-button')
+    expect(footerBtns.length).toEqual(2)
+    expect(footerBtns[0].find('span').element.textContent).toEqual('cancel')
+    expect(footerBtns[1].find('span').element.textContent).toEqual('confirm')
+  })
+
+  test('should append to body, when append-to-body flag is true', async () => {
+    const wrapper = mount({
+      data: () => ({ visible: false }),
+      render() {
+        return (
+          <Drawer
+            title={title}
+            modelValue={this.visible}
+            append-to-body={true}
+          />
+        )
+      },
+    })
+
+    const vm = wrapper.vm as any
+
+    vm.visible = true
+    await nextTick()
+    await rAF()
+    await nextTick()
+    expect(document.querySelector('.el-overlay')?.parentNode).toEqual(
+      document.body
+    )
+  })
+
+  test('should open and close drawer properly', async () => {
+    const onClose = vi.fn()
+    const onClosed = vi.fn()
+    const onOpened = vi.fn()
+
+    const wrapper = mount({
+      data: () => ({ visible: false }),
+      render() {
+        return (
+          <Drawer
+            title={title}
+            modelValue={this.visible}
+            onClose={onClose}
+            onClosed={onClosed}
+            onOpened={onOpened}
+          />
+        )
+      },
+    })
+
+    const vm = wrapper.vm as any
+    await nextTick()
+    await rAF()
+    await nextTick()
+    expect(onOpened).not.toHaveBeenCalled()
+
+    const drawerEl = wrapper.find('.el-overlay').element as HTMLDivElement
+    expect(drawerEl.style.display).toEqual('none')
+
+    vm.visible = true
+    await nextTick()
+    await rAF()
+    expect(drawerEl.style.display).not.toEqual('none')
+    expect(onOpened).toHaveBeenCalled()
+  })
+
+  test('should destroy every child after drawer was closed when destroy-on-close flag is true', async () => {
+    const wrapper = mount(() => (
+      <Drawer
+        ref="drawer"
+        title={title}
+        modelValue={true}
+        append-to-body={false}
+        destroy-on-close={true}
+      >
+        <span>{content}</span>
+      </Drawer>
+    ))
+    await nextTick()
+    await rAF()
+    await nextTick()
+    expect(wrapper.find('.el-drawer__body span').element.textContent).toEqual(
+      content
+    )
+
+    await wrapper.find('.el-drawer__close-btn').trigger('click')
+    await nextTick()
+    await rAF()
+    await nextTick()
+    expect(wrapper.find('.el-drawer__body').exists()).toBe(false)
+  })
+
+  test('should close dialog by clicking the close button', async () => {
```

Comment:
## Questions

This part seems a little different from the previous test. The previous test checks if the state is changed from false to true through the close button, the test you provide checks the state from false to false again.

what do you think about it?